server/src/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -1508,19 +1519,33 @@ public boolean restartVpc(final long vpcId, final boolean cleanUp, final boolean
 
                 // Change the VPC in order to get it updated after the end of
                 // the restart procedure.
-                _vpcDao.update(vpc.getId(), entity);
+                if (_vpcDao.update(vpc.getId(), entity)) {
+                    vpc = entity;
+                }
 
                 // If the offering and redundant column are changing, force the
                 // clean up.
                 forceCleanup = true;
             }
 
             if (forceCleanup) {
```

Comment:
Can you extract the body of this IF to a method?

Reply:
The body of IF is rather small already?

Reply:
Yes it is, if you consider only the body of the IF, but eh method is already quite big.  The more we extract, the more unit tests we enable. Moreover, it becomes easier to read the code.

Reply:
@rafaelweingartner if you look again I did refactor the code to `rollingRestartVpc` I'll stop at that. The lines of code in the `if` block are only 5.

Reply:
I noticed it. This comment is the one I posted before you refactored the code. Github maintained the comment even though you have changed part of this code.